src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -0,0 +1,332 @@
+      if (widget == null) {
+        widget = Shuffleboard.getTab(getName()).add("Pivot Arm", mech);
+      }
+      simSensor = new SimDeviceSim("TOF", CANIDs.kConveyorSensor);
+      simbeam = simSensor.getBoolean("BeamBroken");
+    }
+  }
+
+  public Trigger getBeamBroken() {
+    return sensor.beamBroken();
+  }
+
+  private boolean atSetpoint() {
+    return pivotMotor.getPosition().getValue().isNear(pivotSetpoint, kPivotTolerance);
+  }
+
+  private void setPivotMotor(double speed) {
+    pivotMotor.set(speed);
+  }
+
+  private void setConveyerMotor(double speed) {
+    conveyorMotor.set(speed);
+  }
+
+  private void setIntakeMotor(double speed) {
+    intakeMotor.set(speed);
+  }
+
+  public Angle getPivotPosition() {
+    return pivotMotor.getPosition().getValue();
+  }
+
+  /** Pivot down */
+  public Command extendPivotCommand() {
+    return runOnce(
+        () -> {
+          pivotMotor.setControl(new MotionMagicVoltage(kPivotExtendAngle));
+          pivotSetpoint = kPivotExtendAngle;
+        });
+  }
+
+  /** Pivot up */
+  public Command retractPivotCommand() {
+    return runOnce(
+        () -> {
+          pivotMotor.setControl(new MotionMagicVoltage(kPivotRetractAngle));
+          pivotSetpoint = kPivotRetractAngle;
+        });
+  }
+
+  // Made a command to spin clockwise
+  public Command intakeCommand() {
+    return startEnd(() -> setIntakeMotor(kIntakeSpeed), () -> setIntakeMotor(0));
+  }
+
+  // Made a command to spin counter clockwise
+  public Command outtakeCommand() {
+    return startEnd(() -> setIntakeMotor(-kIntakeSpeed), () -> setIntakeMotor(0));
+  }
+
+  /** Pivots down when user presses button */
+  public Command pivotDownCommand() {
+    return startEnd(() -> setPivotMotor(-kPivotSpeed), () -> setPivotMotor(0));
+  }
+
+  /** Pivots up when user presses button */
+  public Command pivotUpCommand() {
+    return startEnd(() -> setPivotMotor(kPivotSpeed), () -> setPivotMotor(0));
+  }
+
+  /** Pushes game piece from conveyor into birdhouse */
+  public Command conveyorFeed() {
+    return startEnd(() -> setConveyerMotor(-kConveyorSpeed), () -> setConveyerMotor(0));
+  }
+
+  public Command conveyorEject() {
+    return startEnd(() -> setConveyerMotor(kConveyorSpeed), () -> setConveyerMotor(0));
+  }
+
+  public Command intakeAndConveyorCommand() {
+    return Commands.run(
+        () -> {
+          setConveyerMotor(kConveyorSpeed);
+          setIntakeMotor(kIntakeSpeed / 5.0);
+        });
+  }
+
+  public Command intakeToBirdhousePhase1() {
+    return startEnd(
+            () -> {
+              extendPivotCommand().initialize();
+              intakeCommand().initialize();
+              conveyorFeed().initialize();
+            },
+            () -> {
+              extendPivotCommand().end(false);
+              intakeCommand().end(false);
+              conveyorFeed().end(false);
+            })
+        .until(sensor.beamBroken());
+  }
```

Comment:
I believe you want a runEnd here. startEnd will only run this once, and not until the beam break. Also this command has some added complexity because the scorer has the distance sensor that your looking for, this sensor is that the end of the intake, so if it's broken then it will just stop the coral in the intake. This logic will most likely have to bee done in RobotContainer. 
I also don't think you need to phases for this.

Reply:
startEnd is correct here. 
The scorer has the distance sensor? Isn't it the LaserCAN sensor implemented in this class?

Reply:
There are 3 Distance sensor on the intake so that we know were the coral is on the intake, but for feeding it to the Scorer, we have another TOF to now the coral is there. For this use case, I believe we would want to use that sensor.

Reply:
I'd like to get this code into main, so lets push this controls issue until after we get this code into main and can more easily iterate on it.